src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowTransformerControlTest.java

```diff
@@ -0,0 +1,215 @@
+                .setRatedU2(33.0)
+                .setR(17.0)
+                .setX(10.0)
+                .setG(0.00573921028466483)
+                .setB(0.000573921028466483)
+                .setBus1("BUS_2")
+                .setBus2("BUS_3")
+                .add();
+
+        twt.newRatioTapChanger()
+                .beginStep()
+                .setRho(0.9)
+                .setR(0.1089)
+                .setX(0.01089)
+                .setG(0.8264462809917356)
+                .setB(0.08264462809917356)
+                .endStep()
+                .beginStep()
+                .setRho(1.0)
+                .setR(0.121)
+                .setX(0.0121)
+                .setG(0.8264462809917356)
+                .setB(0.08264462809917356)
+                .endStep()
+                .beginStep()
+                .setRho(1.1)
+                .setR(0.1331)
+                .setX(0.01331)
+                .setG(0.9090909090909092)
+                .setB(0.09090909090909092)
+                .endStep()
+                .setTapPosition(0)
+                .setLoadTapChangingCapabilities(true)
+                .setRegulating(false)
+                .setTargetV(33.0)
+                .setRegulationTerminal(load3.getTerminal())
+                .add();
+
+        return network;
+    }
+
+    @BeforeEach
+    void setUp() {
+
+        network = createNetwork();
+        loadFlowRunner = new LoadFlow.Runner(new OpenLoadFlowProvider(new DenseMatrixFactory()));
+        parameters = new LoadFlowParameters();
+        parameters.setTransformerVoltageControlOn(false);
+        OpenLoadFlowParameters parametersExt = new OpenLoadFlowParameters()
+                .setSlackBusSelector(new FirstSlackBusSelector())
+                .setDistributedSlack(false);
+        this.parameters.addExtension(OpenLoadFlowParameters.class, parametersExt);
+    }
+
+    @Test
+    void baseCaseTest() {
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(134.273, bus2);
+        assertVoltageEquals(27.0038, bus3);
+    }
+
+    @Test
+    void tapPlusTwoTest() {
+        twt.getRatioTapChanger().setTapPosition(2);
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        assertTrue(result.isOk());
+
+        assertVoltageEquals(135.0, bus1);
+        assertVoltageEquals(134.281, bus2);
+        assertVoltageEquals(34.427, bus3);
+    }
+
+    @Test
+    void voltageControlTest() {
+        parameters.setTransformerVoltageControlOn(true);
+        twt.getRatioTapChanger()
+                .setTargetDeadband(0)
+                .setRegulating(true)
+                .setTapPosition(0)
+                .setRegulationTerminal(twt.getTerminal2())
+                .setTargetV(34.0);
+
+        LoadFlowResult result = loadFlowRunner.run(network, parameters);
+        // assertTrue(result.isOk()); FIXME
```

Comment:
@geofjamg it is just a first version of voltage control by transformer and it is not working. At this test, the norm of my cost function is ````NaN````. Do you have any idea of the problem ?

Reply:
This is because the way tap changer voltage target equation is defined in AcEquationSystem.createBranchVoltageTargetEquation.
You should define a new equation type for tap changer target voltage (like BUS_TRANSFO_V)) because if you look at Equation.initTargets for type BUS_V, we look for a generator target voltage and not a tap changer target voltage so result is nan and this nan is added to targets vector before first NR.